rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/DatasetService.scala

```diff
@@ -114,8 +123,24 @@ class DatasetService @Autowired()(datasetMongoRepository: DatasetMongoRepository
     }
   }
 
-  def replaceProperties(username: String, datasetName: String,
-                        updatedProperties: Option[Map[String, String]]): Future[Option[Dataset]] = {
+  def updateProperties(username: String, datasetName: String, datasetVersion: Int,
+                       updatedProperties: Map[String, String]): Future[Option[Dataset]] = {
+    for {
+      s <- validateProperties(updatedProperties).flatMap {
+        case validation if !validation.isValid => Future.failed(ValidationException(validation)) // warnings are ok for update
```

Comment:
Maybe warnings could be logged?

Reply:
I am considering doing even a better thing -> to communicate it back to the caller as payload.

Reply:
Done.